central/complianceoperator/v2/report/manager/watcher/scanwatcher.go

```diff
@@ -0,0 +1,239 @@
+package watcher
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	complianceIntegrationDS "github.com/stackrox/rox/central/complianceoperator/v2/integration/datastore"
+	scanDS "github.com/stackrox/rox/central/complianceoperator/v2/scans/datastore"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/set"
+	"golang.org/x/mod/semver"
+)
+
+var (
+	log                                                = logging.LoggerForModule()
+	ScanTimeoutError                                   = errors.New("scan watcher timed out")
+	ComplianceOperatorNotInstalledError                = errors.New("compliance operator is not installed")
+	ComplianceOperatorVersionError                     = errors.New("compliance operator version")
+	ComplianceOperatorIntegrationDataStoreError        = errors.New("unable to retrieve compliance operator integration")
+	ComplianceOperatorIntegrationZeroIntegrationsError = errors.New("no compliance operator integrations retrieved")
+)
+
+const (
+	minimumComplianceOperatorVersion = "v1.6.0"
+	checkCountAnnotationKey          = "compliance.openshift.io/check-count"
+	lastScannedAnnotationKey         = "compliance.openshift.io/last-scanned-timestamp"
+	defaultChanelSize                = 100
+	defaultTimeout                   = 10 * time.Minute
+)
+
+// ScanWatcher determines if a scan is running or has completed.
+type ScanWatcher interface {
+	PushScan(v2 *storage.ComplianceOperatorScanV2) error
+	PushCheckResult(*storage.ComplianceOperatorCheckResultV2) error
+	Finished() concurrency.ReadOnlySignal
+	Stop()
+}
+
+// ScanWatcherResults is returned when the watcher detects that the scan is completed.
+type ScanWatcherResults struct {
+	Ctx          context.Context
+	WatcherID    string
+	Scan         *storage.ComplianceOperatorScanV2
+	CheckResults set.StringSet
+	Error        error
+}
+
+// IsComplianceOperatorHealthy indicates whether Compliance Operator is ready for automatic reporting
+func IsComplianceOperatorHealthy(ctx context.Context, clusterID string, complianceIntegrationDataStore complianceIntegrationDS.DataStore) error {
+	coStatus, err := complianceIntegrationDataStore.GetComplianceIntegrationByCluster(ctx, clusterID)
+	if err != nil {
+		return errors.Wrap(err, ComplianceOperatorIntegrationDataStoreError.Error())
+	}
+	if len(coStatus) == 0 {
+		log.Errorf("No compliance integrations retrieved from cluster %s", clusterID)
+		return ComplianceOperatorIntegrationZeroIntegrationsError
+	}
+	if !coStatus[0].GetOperatorInstalled() {
+		return ComplianceOperatorNotInstalledError
+	}
+	if semver.Compare(coStatus[0].GetVersion(), minimumComplianceOperatorVersion) < 0 {
+		return ComplianceOperatorVersionError
+	}
+	return nil
+}
```

Comment:
I was thinking about what you said in the call that this is unused - maybe we shall then move it to the PR where it is used? If no such PR exists yet, then this should have at least a comment or test coverage (if possible).